tests/basic.rs

```diff
@@ -16,9 +17,7 @@ fn test_cat_blocking() {
     assert_eq!(output.next().unwrap(), "foo\r\n");
     assert_eq!(output.next().unwrap(), "foo\r\n");
 
-    pty.write_all(&[4u8]).unwrap();
-    let status = child.wait().unwrap();
-    assert_eq!(status.code().unwrap(), 0);
+    child.kill().unwrap();
```

Comment:
On macos, this `^D` doesn't seem to kill the cat. :shrug: ??

If we care, we better figure out why. If we don't, we may as well allow the child to be dropped implicitly, I suppose.

Reply:
i'd rather figure out what's going on here - `^D` does work to terminate a `cat` process when run normally on macos, so this crate (or test) is doing something wrong here, and i don't want to just sidestep that.

Reply:
i think this one is actually related to your change above - you're keeping the pts alive longer than before, which apparently prevents the wait from succeeding. if you instead do something like
```rust
    let mut child = {
        let pts = pty.pts().unwrap();
        pty.resize(pty_process::Size::new(24, 80)).unwrap();
        pty_process::blocking::Command::new("cat")
            .spawn(&pts)
            .unwrap()
    };
```
`^D` should work again.

Reply:
Hrm... perhaps. Assuming this is true, wouldn't a user be prone to making a similar mistake such as we made in this test?

Reply:
When spawning a new process, the `Pts` is borrowed. Is this because it's meant for reuse? If not, can it instead by passed by value so that `spawn()` can drop it?

Reply:
Pts is meant for reuse (probably in UNIX in general) as demonstrated in

https://github.com/doy/pty-process/blob/74c223df6dd285f17108c8f442d71fd5ddb9561e/tests/behavior.rs#L3-L29

So, alternatively, what does dropping the Pts do... or what does not dropping it do to prevent the `cat` child process from exiting on a signal?

Reply:
It seems that blocking::Pts doesn't have a Drop impl and neither does sys::Pts. It would be std::os::fd::OwnedFd that needs to be dropped in order for cat to terminate on a signal. On macos, at least.

Which kind of makes me think that it may be that cat is terminating and something else does not. But that's just a hunch. We'd have to take a closer look. We can start with the wait method.

Reply:
This library's spawn method returns a plain std::process::Child. So it is  std::process::Child::wait that is blocking forever even though a signal was sent to cat.

One option: cat is not receiving the signal.
Another option: cat is receiving the signal and terminating—what else would it do?—but for another reason the wait method keeps blocking. Something that has to do with that non-dropped Pts.